src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -323,66 +300,36 @@ private void transitionCaseGroupStatus(final Case targetCase, final CaseEvent ca
-      final CaseEvent caseEvent,
-      final Case newCase)
+  private void processActionPlanChange(final Case targetCase, final boolean enrolments)
       throws CTPException {
 
+    List<CaseGroup> caseGroups =
+        caseGroupService.findCaseGroupsForExecutedCollectionExercises(targetCase);
+
     for (CaseGroup caseGroup : caseGroups) {
 
       // fetch all B and BI cases associated to the case group being processed
       List<Case> cases =
           caseRepo.findByCaseGroupFKOrderByCreatedDateTimeDesc(caseGroup.getCaseGroupPK());
 
-      // Create a new BI case for the respondent enrolling (if one doesn't already exit)
-      // This is primarily to guard against multiple RESPONDENT_ENROLED case events for the same
-      // user
-      // caused by a double click scenario in the verification email journey
-      List<Case> biCases =
-          cases
-              .stream()
-              .filter(
-                  c ->
-                      c.getSampleUnitType().toString().equals("BI")
-                          && c.getPartyId().equals(newCase.getPartyId()))
-              .collect(Collectors.toList());
-      if (biCases.size() != 0) {
-        log.warn(
-            "Existing BI case found during enrolment for "
-                + "partyid: {} for casegroup: {} with caseid: {}",
-            newCase.getPartyId().toString(),
-            caseGroup.getId(),
-            biCases.get(0).getId());
-      } else {
-        Case c = new Case();
-        c.setPartyId(newCase.getPartyId());
-        createNewCase(category, caseEvent, cases.get(0), c);
-        log.info(
-            "BI case created during enrolment for partyid: {} for casegroup: {}",
-            newCase.getPartyId().toString(),
-            caseGroup.getId());
+      List<ActionPlanDTO> actionPlans =
+          actionSvcClientService.getActionPlans(caseGroup.getCollectionExerciseId(), enrolments);
+
+      if (actionPlans == null || actionPlans.size() != 1) {
+        log.error(
+            "One action plan expected for collectionExerciseId={} with active enrolment",
+            caseGroup.getCollectionExerciseId());
+        throw new IllegalStateException(
+            "Expected one action plan for collection exercise with active enrolments");
       }
 
-      // Transition each of the B cases for the casegroup being enrolled for
-      List<Case> caseTypeBs =
-          cases
-              .stream()
-              .filter(c -> c.getSampleUnitType().toString().equals("B"))
-              .collect(Collectors.toList());
-      for (Case caseTypeB : caseTypeBs) {
-        effectTargetCaseStateTransition(category, caseTypeB);
+      for (Case caze : cases) {
+        if (caze.getSampleUnitType() == SampleUnitType.B) {
+          caze.setActionPlanId(actionPlans.get(0).getId());
+          caseRepo.saveAndFlush(caze);
+          notificationPublisher.sendNotification(
+              prepareCaseNotification(caze, CaseDTO.CaseEvent.ACTIONPLAN_CHANGED));
```

Comment:
We still need to add the functionality in action to handle this kind of notification